js/fileinput.js

```diff
@@ -124,6 +130,93 @@
     }
   },
 
+  Fileinput.prototype.setImageTransform = function($img, file) {
+      var Fileinput = this;
+      var reader = new FileReader();
+      reader.onload = function(me) {
+        var transform = false;
+        var view = new DataView(reader.result);
+        var exif = Fileinput.getImageExif(view);
+        if (exif) {
+            Fileinput.resetOrientation($img, exif);
+        }
+      }
+
+      reader.readAsArrayBuffer(file);
+  }
+
+  Fileinput.prototype.getImageExif = function(view) {
+    if (view.getUint16(0, false) != 0xFFD8) { 
+      return -2;
+    }    
+    var length = view.byteLength, offset = 2;
+    while (offset < length) {
+      var marker = view.getUint16(offset, false);
+          offset += 2;
+      if (marker == 0xFFE1) {
+        if (view.getUint32(offset += 2, false) != 0x45786966) { 
+          return -1;
+        }
+        var little = view.getUint16(offset += 6, false) == 0x4949;
+            offset += view.getUint32(offset + 4, little);
+        var tags = view.getUint16(offset, little);
+            offset += 2;
+        for (var i = 0; i < tags; i++)   {
+          if (view.getUint16(offset + (i * 12), little) == 0x0112) {
+            return view.getUint16(offset + (i * 12) + 8, little); 
+          }
+        }
+      }
+      else if ((marker & 0xFF00) != 0xFF00){
+         break;
+      } else {
+        offset += view.getUint16(offset, false);
+      } 
+    }
+
+    return -1;
+  }
+
+  Fileinput.prototype.resetOrientation = function($img, transform) {
+  var img = new Image();    
+  
+  img.onload = function() {
+    var width = img.width,
+        height = img.height,
+        canvas = document.createElement('canvas'),
+        ctx = canvas.getContext("2d");
```

Comment:
I prefer rotating using CSS, rather than using canvas.

```css
.fileinput-flip-horizontal {
    -ms-transform: scaleX(-1);
    -webkit-transform: scaleX(-1);
    transform: scaleX(-1);
    filter: FlipH;
}
.fileinput-flip-vertical {
    -ms-transform: scaleY(-1);
    -webkit-transform: scaleY(-1);
    transform: scaleY(-1);
    filter: FlipY;
}
.fileinput-rotate-left {
    -ms-transform: rotate(-90deg);
    -webkit-transform: rotate(-90deg);
    transform: rotate(-90deg);
    filter:progid:DXImageTransform.Microsoft.BasicImage(rotation=3);
}
.fileinput-rotate-right {
    -ms-transform: rotate(90deg);
    -webkit-transform: rotate(90deg);
    transform: rotate(90deg);
    filter:progid:DXImageTransform.Microsoft.BasicImage(rotation=1);
}
.fileinput-rotate-full {
    -ms-transform: rotate(180deg);
    -webkit-transform: rotate(180deg);
    transform: rotate(180deg);
    filter:progid:DXImageTransform.Microsoft.BasicImage(rotation=2);
}
.fileinput-flip-horizontal.fileinput-rotate-left {
    -ms-transform: scaleX(-1) rotate(-90deg);
    -webkit-transform: scaleX(-1) rotate(-90deg);
    transform: scaleX(-1) rotate(-90deg);
    filter: FlipH progid:DXImageTransform.Microsoft.BasicImage(rotation=3);
}
.fileinput-flip-horizontal.fileinput-rotate-right {
    -ms-transform: scaleX(-1) rotate(90deg);
    -webkit-transform: scaleX(-1) rotate(90deg);
    transform: scaleX(-1) rotate(90deg);
    filter: FlipH progid:DXImageTransform.Microsoft.BasicImage(rotation=1);
}
.fileinput-flip-vertical.fileinput-rotate-left {
    -ms-transform: scaleY(-1) rotate(-90deg);
    -webkit-transform: scaleY(-1) rotate(-90deg);
    transform: scaleY(-1) rotate(-90deg);
    filter: FlipY progid:DXImageTransform.Microsoft.BasicImage(rotation=3);
}
.fileinput-flip-vertical.fileinput-rotate-right {
    -ms-transform: scaleY(-1) rotate(90deg);
    -webkit-transform: scaleY(-1) rotate(90deg);
    transform: scaleY(-1) rotate(90deg);
    filter: FlipY progid:DXImageTransform.Microsoft.BasicImage(rotation=1);
}
```